### PR TITLE
Add SPI support

### LINKED ISSUE #25: Add SPI peripheral
Tracking issue for adding a SPI master implementation for the Teensy 4. The steps below are influenced by the work needed to bring-up an I2C master:

- [x] Add the `LPSPI1` PAC subcrate to `imxrt1062-pac`.
  - If it's not already available, [build the PAC supercrate](https://github.com/mciantyre/teensy4-rs/blob/master/svd/README.md), from which we'll import the `LPSPI1` module as its own crate.
  - Import the module as its own crate using the import tool. Depends on #24.
- [x] Add IOMUXC pin configurations. It might look similar to some of the I2C work in 9dd4e90 and 26cfd09.
- [ ] Initialize SPI module (clocks, pin configurations, etc.)
- [ ] Implement `embedded-hal` traits
- [ ] Add docs
  - Update the pin muxing table in the `teensy4-bsp` docs
  - Document the module / API
- [ ] Add an example to the `teensy4-examples`. The behavior should be verifiable by someone with a logic analyzer.

### PR DESCRIPTION
**Pull request by dflemstr**
This PR is currently a work-in-progress.

I'm trying to add SPI support to the HAL crate; I have a couple of Teensy 4.0's to test with.  However, since the SPI support probably involves figuring out some DMA stuff, I'm very happy to collaborate with someone on this who is more knowledgeable

Fixes #25. (eventually)

### CODE REVIEW

**Comment by mciantyre** [comment]
Thanks for starting this!

Happy to help if you would like to go straight with a DMA-based approach. I was taking a look at the auto-generated code in the `imxrt1062-dma0` crate. It looks like every single DMA transfer control descriptor is broken out into its own module, which might make it tricky to use. The TCD registers are the same, no matter the DMA channel, so the specialization seems unnecessary. We should be able to simplify it, either by modifying the SVD (preferred), or by modifying the sub-PAC and maintaining it outside of auto-generated code workflow. I can work on that if you'd like, and then we can implement the SPI device atop that feature.

If we'd like to hold off on DMA, that's OK with me too. It looks like [the default Teensy 4 libraries don't use DMA](https://github.com/PaulStoffregen/SPI/blob/master/SPI.h#L1221). But, there's [an option to use DMA-based transfer](https://github.com/PaulStoffregen/SPI/blob/master/SPI.h#L1248) if you have an event responder. An implementation of the [`embedded-hal` SPI trait](https://docs.rs/embedded-hal/0.2.3/embedded_hal/spi/trait.FullDuplex.html) might not warrant a DMA implementation, given that we're transferring and receiving individual words. The internal and independent 16 word receive and transfer buffers might suffice.

**Comment by dflemstr** [comment]
@mciantyre I rebased on your recent GPIO changes and had to add some more GPIO pins to make things work.  I also had to change how the LED pin works on teensy4, since pin 13 is also used for the SCK pin of one of the SPI buses.

I will add an example that I can run locally to verify that everything is sound, but I don't have that many generic SPI devices available (only very specific ones for my use case) so I'm not sure if the example will be super reusable...

File: imxrt1062-hal/src/spi.rs [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+    fn new(source_clock: ccm::Frequency, reg: *const pac::lpspi1::RegisterBlock) -> Self {
+        Builder {
+            _module: PhantomData,
+            // Safety: pointer points to static memory
+            reg: unsafe { &*reg },
+            source_clock,
+        }
+    }
+
+    /// Builds an SPI peripheral from the SDO, SDI, SCK and PCS0 pins. The return
```
[the remark this answers is not in the document]
Comment (dflemstr) [reply imxrt1062-hal/src/spi.rs]: Sorry, this was a remnant from when I tried to support both master and slave modes with the same struct. I will update the comment

File: imxrt1062-hal/src/ccm.rs [hunk 2]
```diff
@@ -484,6 +484,50 @@ pub mod i2c {
...
+
+    pub type PrescalarSelect = ccm::cbcmr::LPSPI_PODF_A;
+
+    impl From<ClockSelect> for Frequency {
+        fn from(clock_select: ClockSelect) -> Self {
+            match clock_select {
+                ClockSelect::Pll3Pfd1 => Frequency(664_615_384),
+                ClockSelect::Pll3Pfd0 => Frequency(720_000_000),
+                ClockSelect::Pll2 => Frequency(528_000_000),
+                ClockSelect::Pll2Pfd2 => Frequency(396_000_000),
```
[the remark this answers is not in the document]
Comment (dflemstr) [reply imxrt1062-hal/src/ccm.rs]: I'd prefer to only support PLL2, then :)

File: imxrt1062-hal/src/gpio.rs [hunk 3]
```diff
@@ -123,6 +188,94 @@ macro_rules! ios {
     };
 }
 
+// Bank 1; these are really controlled by IOMUXC_GPR_GPR26 but for some reason I can't access that
+// register here.
```
[the remark this answers is not in the document]
Comment (dflemstr) [reply imxrt1062-hal/src/gpio.rs]: Yup, that works!

**Comment by dflemstr** [comment]
@mciantyre I've been using this example code as an example, note that it is very prototypey.  So far it doesn't work very well, it almost looks like the SPI bus is not configured in 8 bit mode or something, but haven't been able to dig into more details.  Maybe it's easier for you if you have a simpler SPI device to try with.

https://gist.github.com/dflemstr/a51798ef7ec70138ea89c5f574829dc3

As you can see, the code manually configures the CS pin.

**Comment by mciantyre** [comment]
I gave this a try with my `MPU9250`. I needed to make a few changes before I could query the sensor's `WHO_AM_I` register. I commented on the changes in 328d334, which is part of this repo's dflemstr-spi branch. Could you try merging `mciantyre/dflemstr-spi` and testing out the changes in your setup?

**Comment by dflemstr** [comment]
I think your changes look great!

**Comment by mciantyre** [comment]
Thanks again for putting all of this together. If it's working for your use-case, and there's nothing more to do, I'll merge this over the weekend.

**Comment by mciantyre** [comment]
Not something that we need to add now, but I might extend on your work with 16-bit transfer support, and support for the integrated chip select. In a test commit, b945df0, I was noticing much better timing and full duplex reliability when using the internal CS instead of a normal GPIO. I'm happy to tack this on in a later point in time.

**Pull request merged**